codewind-che-sidecar/deploy-pfe/pkg/codewind/deploy_test.go

```diff
@@ -240,3 +240,66 @@ func TestCreatePerformanceService(t *testing.T) {
 		})
 	}
 }
+
+// TestVerifyPFEEnvVars verifies that the environment variables passed into Codewind-PFE have the proper values
+func TestVerifyPFEEnvVars(t *testing.T) {
+	codewindInstance := setupCodewind()
+	tests := []struct {
+		name     string
+		codewind Codewind
+	}{
+		{
+			name:     fmt.Sprintf("Verify PFE Environment Variables"),
+			codewind: codewindInstance,
+		},
+	}
+	for _, tt := range tests {
+		pfeDeploy := createPFEDeploy(tt.codewind)
+		pfeService := createPFEService(tt.codewind)
+
+		performanceService := createPerformanceService(tt.codewind)
+
+		pfeEnvVars := pfeDeploy.Spec.Template.Spec.Containers[0].Env
+
+		for _, env := range pfeEnvVars {
+			// Verify that KUBE_NAMESPACE and TILLER_NAMESPACE matches the namespace that codewind is deployed in
+			if env.Name == "KUBE_NAMESPACE" {
+				if env.Value != pfeDeploy.GetNamespace() {
+					t.Errorf("KUBE_NAMESPACE doesn't match the namespace that Codewind is deployed in %v\n", performanceService.GetName())
+				}
+				continue
+			}
+			if env.Name == "TILLER_NAMESPACE" {
+				if env.Value != pfeDeploy.GetNamespace() {
+					t.Errorf("TILLER_NAMESPACE doesn't match Performance Dashboard service name: %v\n", performanceService.GetName())
```

Comment:
Should the check and error message be referencing some tiller specific metadata rather than `pfeDeploy.GetNamespace()`?

Reply:
Nope. That variable needs to match the namespace we're deploying Codewind in, to ensure the tiller is also running there.